homeassistant/components/lg_soundbar/config_flow.py

```diff
@@ -0,0 +1,76 @@
+"""Config flow to configure the LG Soundbar integration."""
+import temescal
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT, CONF_UNIQUE_ID
+
+from .const import DEFAULT_HOST, DEFAULT_PORT, DOMAIN
+
+DATA_SCHEMA = {
+    vol.Required(CONF_HOST, default=DEFAULT_HOST): str,
+    vol.Required(CONF_PORT, default=DEFAULT_PORT): vol.Coerce(int),
+    vol.Optional(CONF_NAME): str,
+}
+
+
+class LGSoundbarConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """LG Soundbar config flow."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+        if user_input is None:
+            return await self._show_form(user_input)
+        if not errors:
+            try:
+                uuid = self.test_connect(user_input[CONF_HOST], user_input[CONF_PORT])
```

Comment:
You're calling a sync method from async context. Run in executor:
```suggestion
                uuid = await self.hass.async_add_executor_job(self.test_connect, user_input[CONF_HOST], user_input[CONF_PORT])
```

Reply:
nice i didnt know hass had a helper for this.